airbyte-server/src/main/java/io/airbyte/server/apis/ConfigurationApi.java

```diff
@@ -341,17 +341,17 @@ public SourceDefinitionRead getSourceDefinition(final SourceDefinitionIdRequestB
 
   @Override
   public SourceDefinitionRead getSourceDefinitionForWorkspace(final SourceDefinitionIdWithWorkspaceId sourceDefinitionIdWithWorkspaceId) {
-    return null;
+    return execute(() -> sourceDefinitionsHandler.getSourceDefinitionForWorkspace(sourceDefinitionIdWithWorkspaceId));
   }
 
   @Override
   public SourceDefinitionRead createSourceDefinition(final SourceDefinitionCreate sourceDefinitionCreate) {
```

Comment:
I wonder if we should add a comment to this method explaining why we're creating a private definition. It might make sense that we deprecate this route altogether since I think the only API-based definition creation use-case we anticipate is custom, right? Forgive me if we already discussed this at some point, I think from the perspective of an outside reader I could see this being confusing and worthy of a comment/javadoc

Reply:
yeah makes sense.  I don't think we discussed this route specifically.  I guess this could be used for admins to create a private non-custom definition, though there's of course no need for an admin to use such a route add definitions to the db.